entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy

```diff
@@ -365,7 +387,7 @@ declare private function es-codegen:value-for-conversion(
         then $source-entity-type=>map:get("properties")
         else ()
     let $is-missing-source :=
-        (empty($source-properties) or not($target-property-name = map:keys($source-properties)))
+        (exists($source-properties) and ($target-property-name = map:keys($source-properties)))
```

Comment:
Should be `and not`
`(exists($source-properties) and not($target-property-name = map:keys($source-properties)))`